mmit/tests/test_solver.py

```diff
@@ -299,20 +295,22 @@ def test_real_data_4(self):
+        margin = 0.5  # Should not affect the solution
+        values = np.random.rand(1000)
+        moves, preds, costs = solver.compute_optimal_costs(values, values, margin, 1)  # squared hinge
+        np.testing.assert_almost_equal(preds[-1], np.mean(values))
 
-        moves, preds, costs = solver.compute_optimal_costs(target_lower, target_upper, margin, 0)  # linear hinge
-        np.testing.assert_almost_equal(costs[-1], 606.979822, decimal=6)
-        np.testing.assert_almost_equal(preds[-1], 0.309591, decimal=6)
+    def test_uncensored_linear_hinge_yields_median(self):
+        """
+        Uncensored data with linear hinge loss yields median
 
-        moves, preds, costs = solver.compute_optimal_costs(target_lower[::-1].copy(), target_upper[::-1].copy(), margin,
-                                                           0)  # linear hinge
-        np.testing.assert_almost_equal(costs[-1], 606.979822, decimal=6)
-        np.testing.assert_almost_equal(preds[-1], 0.309591, decimal=6)
\ No newline at end of file
+        """
+        margin = 0.5  # Should not affect the solution
+        values = np.random.rand(1000)
+        moves, preds, costs = solver.compute_optimal_costs(values, values, margin, 0)  # linear hinge
+        np.testing.assert_almost_equal(preds[-1], np.median(values))
```

Comment:
great. so our solver can be used as a cummedian function -- not provided with R, but may be useful for other purposes.